.github/workflows/build-push-images-pr.yaml

```diff
@@ -4,15 +4,113 @@ on:
     branches:
       - develop
 jobs:
+  lint:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version:
+          - 3.7
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+
+      - name: Test installing development dependencies
+        working-directory: deployment/docker
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r REQUIREMENTS-dev.txt
+
+      - name: Run Flake8 test
+        run: flake8 .
+
+  test:
+    runs-on: ubuntu-latest
+    needs:
+      - lint
+    env:
+      APP_IMAGE: kartoza/projecta-uwsgi
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Build test image
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: deployment/docker/Dockerfile
+          push: false
+          load: true
+          target: dev
+          tags: ${{ env.APP_IMAGE }}:dev
+          cache-from: |
+            type=gha,scope=test
+            type=gha,scope=prod
+          cache-to: type=gha,scope=test
+
+      - name: Build prod image
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: deployment/docker/Dockerfile
+          push: false
+          load: true
+          target: dev
+          tags: ${{ env.APP_IMAGE }}:prod
+          cache-from: |
+            type=gha,scope=test
+            type=gha,scope=prod
+          cache-to: type=gha,scope=prod
```

Comment:
Similar with above, but now builds the `prod` target as `prod` tag.
It will reuse previous cache above because building `dev` target also build `prod` target as dependencies.